# Assignment: E-commerce Sales Analysis (Day 8)

Create a Data Analysis Notebook using Python and Pandas with the provided e-commerce sales dataset. Load the dataset into a Pandas DataFrame and perform analysis using data selection, filtering, sorting, groupby(), and aggregation functions. Analyze sales based on different categories, cities, products, and payment methods, and calculate statistics such as total sales, average sales, highest and lowest sales, total quantity sold, and number of orders. Apply suitable filters and sorting operations to identify specific orders and top-performing products or categories. Present all operations clearly in a well-organized Colab Notebook and write a few meaningful observations based on your findings. * You can find the .csv file in the "Study Material" tab in the LMS.

In [30]:
import pandas as pd 

In [31]:
df = pd.read_csv('Day8_Ecommerce_Sales_Dataset.csv')
df.head(10)


,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
0,1001,2026-02-27,Kabir Ali,Hyderabad,South,Electronics,Wireless Headphones,3,1499,0,4497.00,3.8,Cash on Delivery
1,1002,2026-01-24,Aditya Verma,Jammu,North,Sports,Running Shoes,1,2799,0,2799.00,4.1,Cash on Delivery
2,1003,2026-04-18,Sara Ahmed,Chennai,South,Electronics,Power Bank,2,1199,15,2038.30,4.6,Cash on Delivery
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
4,1005,2026-03-30,Priya Menon,Jammu,North,Electronics,Smart Watch,1,3299,5,3134.05,4.2,UPI
5,1006,2026-01-21,Manya Rao,Kochi,South,Books,The Alchemist,1,499,10,449.10,4.2,Cash on Delivery
6,1007,2026-01-12,Alina Shah,Jaipur,North,Home & Kitchen,Water Bottle,2,699,0,1398.00,4.1,Debit Card
7,1008,2026-04-27,Aman Kumar,Jammu,North,Clothing,Hoodie,4,1599,5,6076.20,4.3,Credit Card
8,1009,2026-06-05,Aman Kumar,Mumbai,West,Home & Kitchen,Electric Kettle,3,1499,0,4497.00,4.0,Cash on Delivery
9,1010,2026-03-11,Aman Kumar,Kochi,South,Clothing,Jacket,4,2499,10,8996.40,4.7,Credit Card


In [32]:
df.tail()

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
95,1096,2026-01-26,Aditya Verma,Kochi,South,Sports,Dumbbell Set,5,1999,5,9495.25,4.2,Credit Card
96,1097,2026-02-02,Arjun Nair,Mumbai,West,Clothing,T-Shirt,1,799,15,679.15,3.5,UPI
97,1098,2026-01-04,Yusuf Dar,Bengaluru,South,Home & Kitchen,Electric Kettle,4,1499,0,5996.00,4.2,Net Banking
98,1099,2026-03-06,Hiba Sheikh,Ahmedabad,West,Books,Machine Learning Basics,2,999,0,1998.00,3.8,UPI
99,1100,2026-02-08,Kabir Ali,Pune,West,Books,Data Science Handbook,5,899,0,4495.00,4.8,Debit Card


In [33]:
df.shape

(100, 13)

In [34]:
df.columns

Index(['Order_ID', 'Order_Date', 'Customer', 'City', 'Region', 'Category',
       'Product', 'Quantity', 'Unit_Price', 'Discount_Percent', 'Total_Sales',
       'Rating', 'Payment_Method'],
      dtype='object')

In [35]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Order_ID          100 non-null    int64  
 1   Order_Date        100 non-null    object 
 2   Customer          100 non-null    object 
 3   City              100 non-null    object 
 4   Region            100 non-null    object 
 5   Category          100 non-null    object 
 6   Product           100 non-null    object 
 7   Quantity          100 non-null    int64  
 8   Unit_Price        100 non-null    int64  
 9   Discount_Percent  100 non-null    int64  
 10  Total_Sales       100 non-null    float64
 11  Rating            100 non-null    float64
 12  Payment_Method    100 non-null    object 
dtypes: float64(2), int64(4), object(7)
memory usage: 10.3+ KB


In [36]:
df.describe()

,Order_ID,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating
count,100.000000,100.000000,100.000000,100.000000,100.000000,100.000000
mean,1050.500000,2.940000,1647.000000,6.500000,4472.629500,4.322000
std,29.011492,1.391315,970.054669,6.092718,3324.306691,0.414285
min,1001.000000,1.000000,499.000000,0.000000,449.100000,3.500000
25%,1025.750000,2.000000,799.000000,0.000000,1948.050000,4.100000
50%,1050.500000,3.000000,1499.000000,5.000000,3228.300000,4.300000
75%,1075.250000,4.000000,2499.000000,11.250000,6427.562500,4.700000
max,1100.000000,5.000000,3499.000000,15.000000,13995.000000,5.000000


In [37]:

df[['Product', 'Category', 'Total_Sales']]
df.iloc[0:5, 0:3]  

,Order_ID,Order_Date,Customer
0,1001,2026-02-27,Kabir Ali
1,1002,2026-01-24,Aditya Verma
2,1003,2026-04-18,Sara Ahmed
3,1004,2026-03-13,Kabir Ali
4,1005,2026-03-30,Priya Menon


In [38]:
high_sales = df[df['Total_Sales'] > 1000]
high_sales

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
0,1001,2026-02-27,Kabir Ali,Hyderabad,South,Electronics,Wireless Headphones,3,1499,0,4497.00,3.8,Cash on Delivery
1,1002,2026-01-24,Aditya Verma,Jammu,North,Sports,Running Shoes,1,2799,0,2799.00,4.1,Cash on Delivery
2,1003,2026-04-18,Sara Ahmed,Chennai,South,Electronics,Power Bank,2,1199,15,2038.30,4.6,Cash on Delivery
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
4,1005,2026-03-30,Priya Menon,Jammu,North,Electronics,Smart Watch,1,3299,5,3134.05,4.2,UPI
...,...,...,...,...,...,...,...,...,...,...,...,...,...
94,1095,2026-02-25,Maryam Khan,Mumbai,West,Electronics,Power Bank,2,1199,0,2398.00,4.6,Debit Card
95,1096,2026-01-26,Aditya Verma,Kochi,South,Sports,Dumbbell Set,5,1999,5,9495.25,4.2,Credit Card
97,1098,2026-01-04,Yusuf Dar,Bengaluru,South,Home & Kitchen,Electric Kettle,4,1499,0,5996.00,4.2,Net Banking
98,1099,2026-03-06,Hiba Sheikh,Ahmedabad,West,Books,Machine Learning Basics,2,999,0,1998.00,3.8,UPI


In [39]:
city = df[df['City'] == "Jammu"]
city

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
1,1002,2026-01-24,Aditya Verma,Jammu,North,Sports,Running Shoes,1,2799,0,2799.00,4.1,Cash on Delivery
4,1005,2026-03-30,Priya Menon,Jammu,North,Electronics,Smart Watch,1,3299,5,3134.05,4.2,UPI
7,1008,2026-04-27,Aman Kumar,Jammu,North,Clothing,Hoodie,4,1599,5,6076.20,4.3,Credit Card
44,1045,2026-03-27,Reyansh Jain,Jammu,North,Books,Machine Learning Basics,1,999,0,999.00,4.0,Debit Card
81,1082,2026-06-23,Rahul Das,Jammu,North,Sports,Yoga Mat,3,899,15,2292.45,4.5,Credit Card
82,1083,2026-06-25,Priya Menon,Jammu,North,Sports,Football,1,799,5,759.05,3.8,Cash on Delivery


In [40]:
city_with_Category = df[(df["City"] == "Pune") & (df["Category"] == "Sports")]
city_with_Category

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
37,1038,2026-03-08,Aditya Verma,Pune,West,Sports,Dumbbell Set,5,1999,5,9495.25,5.0,Debit Card
59,1060,2026-04-24,Karan Joshi,Pune,West,Sports,Dumbbell Set,3,1999,10,5397.30,5.0,Credit Card


In [41]:
customer_rating = df[df["Rating"] > 4]
customer_rating

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
1,1002,2026-01-24,Aditya Verma,Jammu,North,Sports,Running Shoes,1,2799,0,2799.00,4.1,Cash on Delivery
2,1003,2026-04-18,Sara Ahmed,Chennai,South,Electronics,Power Bank,2,1199,15,2038.30,4.6,Cash on Delivery
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
4,1005,2026-03-30,Priya Menon,Jammu,North,Electronics,Smart Watch,1,3299,5,3134.05,4.2,UPI
5,1006,2026-01-21,Manya Rao,Kochi,South,Books,The Alchemist,1,499,10,449.10,4.2,Cash on Delivery
...,...,...,...,...,...,...,...,...,...,...,...,...,...
92,1093,2026-05-16,Aisha Mir,Hyderabad,South,Sports,Cricket Bat,4,2499,15,8496.60,4.5,Cash on Delivery
94,1095,2026-02-25,Maryam Khan,Mumbai,West,Electronics,Power Bank,2,1199,0,2398.00,4.6,Debit Card
95,1096,2026-01-26,Aditya Verma,Kochi,South,Sports,Dumbbell Set,5,1999,5,9495.25,4.2,Credit Card
97,1098,2026-01-04,Yusuf Dar,Bengaluru,South,Home & Kitchen,Electric Kettle,4,1499,0,5996.00,4.2,Net Banking


In [42]:
payment_counts = df['Payment_Method'].value_counts()
payment_counts 

Payment_Method
UPI                 26
Credit Card         22
Debit Card          20
Net Banking         17
Cash on Delivery    15
Name: count, dtype: int64

In [43]:
Products = df['Product'].value_counts()
Products

Product
Wireless Headphones        7
Coffee Maker               7
The Alchemist              6
Yoga Mat                   6
Atomic Habits              6
Water Bottle               5
Running Shoes              5
Cricket Bat                5
Dumbbell Set               5
T-Shirt                    5
Jacket                     5
Electric Kettle            5
Machine Learning Basics    4
Mixer Grinder              4
Football                   4
Python Programming         4
Mechanical Keyboard        3
Power Bank                 3
Smart Watch                3
Wireless Mouse             3
Hoodie                     2
Jeans                      2
Data Science Handbook      1
Name: count, dtype: int64

In [44]:
category = df["Category"].unique()
category

array(['Electronics', 'Sports', 'Books', 'Home & Kitchen', 'Clothing'],
      dtype=object)

In [15]:
sorted_multicol = df.sort_values(by=['Category', 'Total_Sales'], ascending=[True, False])
sorted_multicol

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
99,1100,2026-02-08,Kabir Ali,Pune,West,Books,Data Science Handbook,5,899,0,4495.00,4.8,Debit Card
29,1030,2026-03-13,Karan Joshi,Bengaluru,South,Books,Python Programming,4,749,0,2996.00,4.1,UPI
90,1091,2026-06-26,Hiba Sheikh,Kochi,South,Books,Atomic Habits,5,599,0,2995.00,5.0,Net Banking
87,1088,2026-06-06,Alina Shah,Lucknow,North,Books,Python Programming,4,749,15,2546.60,4.5,Debit Card
72,1073,2026-06-04,Sara Ahmed,Kolkata,East,Books,Atomic Habits,5,599,15,2545.75,4.6,Credit Card
...,...,...,...,...,...,...,...,...,...,...,...,...,...
39,1040,2026-05-18,Aditya Verma,Ahmedabad,West,Sports,Dumbbell Set,1,1999,0,1999.00,4.7,Debit Card
35,1036,2026-06-02,Rohan Mehta,Srinagar,North,Sports,Yoga Mat,2,899,0,1798.00,5.0,Credit Card
20,1021,2026-01-18,Reyansh Jain,Chandigarh,North,Sports,Yoga Mat,1,899,10,809.10,4.7,Credit Card
30,1031,2026-01-04,Rohan Mehta,Ahmedabad,West,Sports,Yoga Mat,1,899,15,764.15,4.1,Credit Card


In [45]:

highest_sales_product = (df.groupby('Product')['Total_Sales'].sum().sort_values(ascending=False).head(1))
print("Highest Revenue Product:\n", highest_sales_product)

Highest Revenue Product:
 Product
Coffee Maker    51610.25
Name: Total_Sales, dtype: float64


In [46]:

highest_sales_category = (df.groupby('Category')['Total_Sales'].sum().sort_values(ascending=False).head(1))
print("Highest Revenue Category:\n", highest_sales_category)

Highest Revenue Category:
 Category
Sports    136026.45
Name: Total_Sales, dtype: float64


In [47]:

cat_stats = df.groupby('Category').agg({'Total_Sales': ['sum', 'mean', 'min', 'max', 'count'],'Quantity': 'sum'})
cat_stats.columns = ['Total_Sales', 'Avg_Sales', 'Min_Sales', 'Max_Sales', 'Num_Orders', 'Total_Qty']
cat_stats

,Total_Sales,Avg_Sales,Min_Sales,Max_Sales,Num_Orders,Total_Qty
Category,,,,,,
Books,36792.40,1752.019048,449.10,4495.00,21,60
Clothing,69028.20,4930.585714,679.15,10620.75,14,39
Electronics,92086.00,4846.631579,1708.10,9496.20,19,57
Home & Kitchen,113329.90,5396.661905,1188.30,12596.40,21,59
Sports,136026.45,5441.058000,759.05,13995.00,25,79


In [48]:

city_stats = df.groupby('City').agg({'Total_Sales': ['sum', 'mean', 'count'],'Quantity': 'sum'})
city_stats.columns = ['Total_Sales', 'Avg_Sales', 'Num_Orders', 'Total_Qty']
city_stats.sort_values('Total_Sales', ascending=False)

,Total_Sales,Avg_Sales,Num_Orders,Total_Qty
City,,,,
Hyderabad,59076.45,8439.492857,7,25
Chennai,47493.20,4749.320000,10,29
Lucknow,46194.65,7699.108333,6,22
Pune,41908.55,4190.855000,10,34
Mumbai,41457.80,5182.225000,8,23
Kochi,37257.50,4139.722222,9,29
Chandigarh,34539.65,4317.456250,8,22
Delhi,30882.80,5147.133333,6,18
Jaipur,25534.00,3647.714286,7,17


In [49]:

payment_stats = df.groupby('Payment_Method').agg({
    'Total_Sales': ['sum', 'mean', 'count']
})
payment_stats.columns = ['Total_Sales', 'Avg_Sales', 'Num_Orders']
payment_stats.sort_values('Total_Sales', ascending=False)

,Total_Sales,Avg_Sales,Num_Orders
Payment_Method,,,
Credit Card,106540.25,4842.738636,22
Debit Card,98064.90,4903.245000,20
Net Banking,96371.25,5668.897059,17
UPI,96163.25,3698.586538,26
Cash on Delivery,50123.30,3341.553333,15


In [50]:

region_stats = df.groupby('Region').agg({'Total_Sales': ['sum', 'mean', 'min', 'max', 'count'],'Quantity': 'sum'})
region_stats.columns = ['Total_Sales', 'Avg_Sales', 'Min_Sales', 'Max_Sales', 'Num_Orders', 'Total_Qty']
region_stats.sort_values('Total_Sales', ascending=False)

,Total_Sales,Avg_Sales,Min_Sales,Max_Sales,Num_Orders,Total_Qty
Region,,,,,,
North,164989.45,4341.827632,449.10,12495.00,38,102
South,157351.45,5245.048333,449.10,13295.25,30,97
West,98155.05,4089.793750,679.15,13995.00,24,69
East,26767.00,3345.875000,899.10,6598.00,8,26


In [53]:

top_five_products = df.groupby('Product')['Total_Sales'].sum().nlargest(5)
top_five_products

Product
Coffee Maker           51610.25
Jacket                 47106.15
Cricket Bat            41483.40
Running Shoes          41285.25
Wireless Headphones    33877.40
Name: Total_Sales, dtype: float64

In [54]:

top_five_categories = df.groupby('Category')['Total_Sales'].sum().nlargest(5)
top_five_categories

Category
Sports            136026.45
Home & Kitchen    113329.90
Electronics        92086.00
Clothing           69028.20
Books              36792.40
Name: Total_Sales, dtype: float64

In [55]:

qty_per_product = df.groupby('Product')['Quantity'].sum().sort_values(ascending=False)
qty_per_product

Product
Wireless Headphones        23
Atomic Habits              21
Jacket                     20
Electric Kettle            19
Cricket Bat                18
The Alchemist              17
Coffee Maker               16
Dumbbell Set               16
Yoga Mat                   16
Running Shoes              15
Water Bottle               15
Football                   14
Wireless Mouse             12
Python Programming         11
Mechanical Keyboard        11
T-Shirt                    10
Mixer Grinder               9
Power Bank                  6
Machine Learning Basics     6
Data Science Handbook       5
Hoodie                      5
Smart Watch                 5
Jeans                       4
Name: Quantity, dtype: int64

## Observations

1.Dataset Size: Total 100 rows and 13 columns inside the data.

2.Data Types: 3 types data only—text (object), decimal numbers (float64), and normal numbers (int64).

3.Column List: Order_ID, Order_Date, Customer, City, Region, Category, Product, Quantity, Unit_Price, Discount_Percent, Total_Sales, Rating, and Payment_Method.

4.Top Category: Sports making most money (1.36 lakh). Home & Kitchen coming second (1.13 lakh), Electronics third (92k), and Books lowest at last (36k).

5.Payment King: UPI used most—26 times. Credit Card 22, Debit Card 20, Net Banking 17, and Cash on Delivery only 15 times.

6.Big Money Item: Coffee Maker single handedly made 51k+ sales. Jacket 47k, Cricket Bat 41k, Running Shoes 41k, and Wireless Headphones 33k.

7.Most Frequent Orders: Both Wireless Headphones and Coffee Maker ordered 7 times each—most popular.

8.Region Scene: North region top on orders (38 orders, 1.64 lakh sales). South had 30 orders but average cart value highest (5,245). West got 24 orders, East very low—only 8 orders.

9.Bulk Buy Items: Dumbbell Set and Data Science Book getting highest quantity in single order (5 pieces each).

10.Customer Feedback: Overall rating very good—average 4.32 out of 5.0 max.

11.Discount Trick: Giving 5% to 15% discount actually making good total money because customer buying more pieces.

12.Total Volume King: Wireless Headphones sold highest overall count—total 23 pieces gone.